src/screens/Wallets/WalletsNew.tsx

```diff
@@ -12,13 +12,42 @@ import {
 import { verticalScale } from '@utils/scaling';
 import { AccountActions, HomeHeader } from './components';
 import { styles } from './styles';
+import { API } from '@api/api';
+import { Transaction } from '@models';
 
 export const HomeScreen = () => {
   const selectedWalletHash = useSelectedWalletHash();
   const { data: account, loading: accountLoading } =
     useCryptoAccountFromHash(selectedWalletHash);
   const usdPrice = useUSDPrice(account?.ambBalance || 0);
 
+  const [tokens, setTokens] = useState<
+    {
+      address: string;
+      name: string;
+      balance: { wei: string; ether: number };
+    }[]
+  >([]);
+
+  const [transactions, setTransactions] = useState<Transaction[]>([]);
+
+  const fetchTokensAndTransactions = async () => {
```

Comment:
It would be more reusable to move it under `hooks` and combine this request with `useQuery`